include/boost/geometry/policies/is_valid/failing_reason_policy.hpp

```diff
@@ -0,0 +1,150 @@
+// Boost.Geometry (aka GGL, Generic Geometry Library)
+
+// Copyright (c) 2015, Oracle and/or its affiliates.
+
+// Contributed and/or modified by Menelaos Karavelas, on behalf of Oracle
+
+// Licensed under the Boost Software License version 1.0.
+// http://www.boost.org/users/license.html
+
+#ifndef BOOST_GEOMETRY_POLICIES_IS_VALID_FAILING_REASON_POLICY_HPP
+#define BOOST_GEOMETRY_POLICIES_IS_VALID_FAILING_REASON_POLICY_HPP
+
+#include <sstream>
+
+#include <boost/geometry/io/dsv/write.hpp>
+#include <boost/geometry/util/range.hpp>
+#include <boost/geometry/algorithms/validity_failure_type.hpp>
+
+
+namespace boost { namespace geometry
+{
+
+
+inline char const* validity_failure_type_message(validity_failure_type failure)
+{
+    switch (failure)
+    {
+    case no_failure:
+        return "Geometry is valid";
+    case failure_few_points:
+        return "Geometry has too few points";
+    case failure_wrong_dimension:
+        return "Geometry has wrong dimension";
+    case failure_not_closed:
+        return "Geometry is defined as closed but is open";
+    case failure_spikes:
+        return "Geometry has spikes";
+    case failure_self_intersections:
+        return "Geometry has invalid self-intersections";
+    case failure_wrong_orientation:
+        return "Geometry has wrong orientation";
+    case failure_holes_outside:
+        return "Geometry has holes defined outside the outer boundary";
+    case failure_nested_holes:
+        return "Geometry has nested holes";
+    case failure_disconnected_interior:
+        return "Geometry has disconnected interior";
```

Comment:
The holes-outside-boundary case refers to the case where an interior boundary is not inside the exterior boundary.
This refers to the case were all holes are inside the (closure of the) exterior boundary, but the interior of the polygon is disconnected. If you look at the example in the documentation of `is_valid`, it refers to exactly this case.